client/verta/tests/unit_tests/test_deployed_model.py

```diff
@@ -0,0 +1,377 @@
+    """ Calling predict with the default settings and getting a 429 results in retries being exhausted. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={},
+        status=429,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    with pytest.raises(RetryError):
+        dm.predict(x=['test_prediction'])
+
+
+def test_predict_http_defaults_status_not_in_retry(mocked_responses) -> None:
+    """ Verify that calling predict with the default settings and getting a response not in `status_forcelist`
+     does not result in retries. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        headers={'verta-request-id': 'hereISaTESTidFROMtheUSER'},
+        json={},
+        status=999,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    dm.predict(x=['test_prediction'])
+    mocked_responses.assert_call_count(PREDICTION_URL, 1)
+
+
+def test_predict_http_default_max_retry_observed(mocked_responses) -> None:
+    """ Calling predict with the default settings and getting a 429 results in retries being exhausted. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={},
+        status=429,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    with pytest.raises(RetryError):
+        dm.predict(x=['test_prediction'])
+        mocked_responses.assert_call_count(PREDICTION_URL, http_session.DEFAULT_MAX_RETRIES + 1)
+        # max_retries + 1 original attempt = total call count
+
+
+def test_predict_with_altered_retry_config(mocked_responses) -> None:
+    """ Calling predict with custom retry parameters changes the retry config and makes the correct requests. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={},
+        status=888,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    with pytest.raises(RetryError):
+        dm.predict(
+            x=['test_prediction'],
+            max_retries=9,
+            retry_status={888},
+            backoff_factor=0.1
+            )
+        mocked_responses.assert_call_count(PREDICTION_URL, 10)
+
+
+def test_predict_with_prediction_id_provided(mocked_responses) -> None:
+    """ Calling predict while providing a value for `prediction_id` updates and includes the headers in the request. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={'test1': 'test1'},
+        status=200,
+        headers={'verta-request-id': 'hereISaTESTidFROMtheUSER'},
```

Comment:
This test's description says

> includes the headers in the request.

but it doesn't look like we're asserting that `dm.predict()` sends the header; I think this argument means to return it in the response.

Based on https://github.com/getsentry/responses/blob/master/README.rst#request-headers-validation, I think we'd need

```suggestion
        match=[matchers.header_matcher({"verta-request-id": "hereISaTESTidFROMtheUSER"})],
```

with

```python
from responses import matchers
```